rasa/utils/tensorflow/models.py

```diff
@@ -223,8 +227,14 @@ def load(cls, model_file_name: Text, *args, **kwargs) -> "RasaModel":
         logger.debug("Loading the model ...")
         # create empty model
         model = cls(*args, **kwargs)
+        # need to train on 1 example to build weights of the correct size
+        model.compile(run_eagerly=True)
+        data_generator = RasaBatchDataGenerator(model_data_example, batch_size=1)
+        model.fit(data_generator, verbose=False)
         # load trained weights
         model.load_weights(model_file_name)
+        # prepare the model for prediction
+        model.prepare_for_predict()
```

Comment:
The call to `prepare_for_predict` is already happening inside `build_for_predict` which is being called by the `_load_model()` of `DIETClassifier` and `TEDPolicy`. I think you should skip making the call here.

Reply:
Ahh no, i now see we have remove `build_for_predict` in this PR.

Reply:
Okay just so you know, we should merge `e2e`/ `master` carefully into this since `model.load()` needs to take care of loading the model in finetune mode or in predict model.